### PR TITLE
Chore: Create plugin does not expect system settings

### PR DESCRIPTION
**Pull request by iLLiCiTiT**
## Changelog Description
System settings are not passed to initialization of create plugin initialization (and `apply_settings`).

## Additional info
Reasons: The system settings are not used anywhere, and since AYON there won't be 2 types of settings.

We need validation of all Create plugins: They should not use `system_settings` in `apply_settings` method and they should not override `__init__` method.

To make this PR safer the system settings can still be passed to creator initialization, but eventually it will be removed in future PR anyway, so it would be just temporary backwards compatibility.

This PR must be updated with latest develop and revalidated changes before merge.

## Testing notes:
1. Technically run all available creators in Publisher

- [x] Traypublisher
- [x] Maya
- [x] Nuke
- [x] AfterEffects
- [x] Photoshop
- [x] 3dsMax
- [x] Houdini
- [ ] TVPaint

### CODE REVIEW

**Comment by BigRoy** [comment]
Unfortunately I did end up using `system_settings` on our end for a Deadline creator plug-in.

See [here](https://github.com/BigRoy/OpenPype/blob/colorbleed_develop3/openpype/modules/deadline/plugins/publish/submit_maya_deadline.py#L164-L172) to use it [here](https://github.com/BigRoy/OpenPype/blob/colorbleed_develop3/openpype/modules/deadline/plugins/publish/submit_maya_deadline.py#L884-L889)

See [here](https://github.com/BigRoy/OpenPype/blob/colorbleed_develop3/openpype/modules/deadline/plugins/publish/collect_pools.py#L34-L39) to use it [here](https://github.com/BigRoy/OpenPype/blob/colorbleed_develop3/openpype/modules/deadline/plugins/publish/collect_pools.py#L61-L75).

Any good ideas on how I could still optimally retrieve our globally define deadline URL - since the project settings URL just picks one by name from the Module Settings? The only thing I can think of is that the `apply_settings`? If I'd initialize a `ModuleManager` I'd still need to do that per plug-in to get the instantiated modules at the time of `apply_settings` so I couldn't get it from a module instance either.

---

Also, since this is a backwards incompatible change for external code bases (e.g. anyone having a custom new style creator using `apply_settings`) it might be good to merge this into a next minor release maybe with a clear note in the changelog?

**Comment by iLLiCiTiT** [comment]
> Does this PR affect those too?

No.

> Looking at the changed files it doesn't but I'm assuming a PR for that might be incoming too?

Publish plugins already have changed code to allow expect only project settings, but it is "less radical" there, because system settings are still used, which won't change until full AYON release.
I've chose more radical approach in create plugins, because I didn't find any usage of them anywhere and couldn't think of the usage...

> Also, since this is a backwards incompatible change for external code bases (e.g. anyone having a custom new style creator using apply_settings) it might be good to merge this into a next minor release maybe with a clear note in the changelog?

I would rather change it to be more backwards compatible than wait for next-minor, last next-minor took more than 6 months.

**Comment by BigRoy** [comment]
For Create plug-ins I'm personally fine with removing it - I'd just need to sync up potential custom creators but if it's clear WHEN I should be doing that then I'm totally fine with it. It's a trivial change.

**Comment by iLLiCiTiT** [comment]
Added backwards compatibility, System settings are still passed to Create plugin. A warning message is logged if `__init__` is overriden or `apply_settings` still expect `system_settings`.

File: openpype/pipeline/create/creator_plugins.py [hunk 1]
```diff
@@ -224,7 +243,8 @@ def identifier(self):
 
         return self.family
 
-    @abstractproperty
+    @property
+    @abstractmethod
```
Comment (BigRoy) [inline openpype/pipeline/create/creator_plugins.py:247]: Isn't this Python 3.3+ only? [src](https://stackoverflow.com/a/48710068)

File: openpype/pipeline/create/creator_plugins.py [hunk 2]
```diff
@@ -208,10 +204,33 @@ def __init__(
...
+            self.apply_settings(project_settings, system_settings)
+
+        init_overriden = self.__class__.__init__ is not BaseCreator.__init__
+        if init_overriden or expect_system_settings:
+            self.log.warning((
+                "WARNING: Source - Create plugin {}."
+                " System settings argument will not be passed to"
+                " '__init__' and 'apply_settings' methods in future versions"
+                " of OpenPype. Planned version to drop the support"
+                " is 3.15.6 or 3.16.0. Please contact Ynput core team if you"
```
Comment (BigRoy) [inline openpype/pipeline/create/creator_plugins.py:224]: The version numbers mentioned here seem low. `develop` is currently on: `3.16.5-nightly.3`?

File: openpype/pipeline/create/creator_plugins.py (hunk 1 above)
Comment (iLLiCiTiT) [reply openpype/pipeline/create/creator_plugins.py:247]: Yes, but this is not Python 2 breaking. I would not bend to Python 2 at this moment.

File: openpype/pipeline/create/creator_plugins.py (hunk 2 above)
Comment (iLLiCiTiT) [reply openpype/pipeline/create/creator_plugins.py:224]: > The version numbers mentioned here seem low. develop is currently on: 3.16.5-nightly.3?

It's 2 patch releases or 1 minor release.

Comment (BigRoy) [reply openpype/pipeline/create/creator_plugins.py:224]: So then it should be `3.16.7 or 3.17.0` right?

Comment (iLLiCiTiT) [reply openpype/pipeline/create/creator_plugins.py:224]: `3.16.5` is not out yet. `3.16.5-nightly.3` is preparation for `3.16.5`.

Comment (iLLiCiTiT) [reply openpype/pipeline/create/creator_plugins.py:224]:
```suggestion
                " is 3.16.6 or 3.17.0. Please contact Ynput core team if you"
```

**Review by jakubjezek001 — APPROVED** [review]
Nuke and TrayPublisher tested and worked normally

**Review by antirotor — APPROVED** [review]
Creators in Maya, Houdini, 3dsmax and Blender works

**Review by kalisp — APPROVED** [review]
Tested in PS, AE, works.

File: openpype/pipeline/create/creator_plugins.py [hunk 3]
```diff
@@ -208,10 +204,33 @@ def __init__(
...
+            self.apply_settings, project_settings
+        ):
+            self.apply_settings(project_settings)
+        else:
+            expect_system_settings = True
+            # Backwards compatibility for system settings
+            self.apply_settings(project_settings, system_settings)
+
+        init_overriden = self.__class__.__init__ is not BaseCreator.__init__
+        if init_overriden or expect_system_settings:
```
Comment (BigRoy) [inline openpype/pipeline/create/creator_plugins.py:218]: Ok, so this warning is now always logged for all classes inheriting from the new `Creator` due to the `Creator` class already overriding `__init__`. :)

**Pull request merged**